include/deal.II/base/function_cspline.h

```diff
@@ -0,0 +1,116 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__function_cspline_h
+#define dealii__function_cspline_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_GSL
+#include <deal.II/base/function.h>
+#include <deal.II/base/point.h>
+#include <gsl/gsl_spline.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Functions
+{
+  DeclException3 (ExcCSplineOrder,
+                  int,
+                  double,
+                  double,
+                  << "CSpline: x[" << arg1 << "] = "<< arg2 <<" >= x["<<(arg1+1)<<" = "<<arg3
```

Comment:
Can you be more verbose about what condition is violated when the given arguments satisfy/don't satisfy these conditions? In essence *also* say in words what is going wrong here.